platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -15,4 +15,36 @@
 [[best_practices_general]]
 = General Gradle Best Practices
 
-Add content here.
+== Do Not Use Internal APIs
+
+Do not use APIs that are inside an `internal` Java package or contain `Internal` or `Impl`
+in the name of the type.
+
+=== Explanation
+
+Using internal APIs is inherently risky and can cause significant problems during upgrades.
+Gradle and many plugins (such as Android Gradle Plugin and Kotlin Gradle Plugin) treat internal
```

Comment:
🤔 Should we call out that sometimes plugins like AGP and KGP have used internals and been bitten by it, and don't copy them?